hugr-py/src/hugr/package.py

```diff
@@ -0,0 +1,130 @@
+"""HUGR package and pointed package interfaces."""
+
+from dataclasses import dataclass, field
+from typing import Generic, TypeVar, cast
+
+import hugr._serialization.extension as ext_s
+from hugr.ext import Extension
+from hugr.hugr.base import Hugr
+from hugr.hugr.node_port import Node
+from hugr.ops import FuncDecl, FuncDefn, Op
+
+__all__ = [
+    "Package",
+    "PackagePointer",
+    "ModulePointer",
+    "ExtensionPointer",
+    "NodePointer",
+    "FuncDeclPointer",
+    "FuncDefnPointer",
+]
+
+
+@dataclass
+class Package:
+    """A package of HUGR modules and extensions.
+
+
+    The HUGRs may refer to the included extensions or those not included.
+    """
+
+    #: HUGR modules in the package.
+    modules: list[Hugr]
+    #: Extensions included in the package.
+    extensions: list[Extension] = field(default_factory=list)
+
+    def _to_serial(self) -> ext_s.Package:
+        return ext_s.Package(
+            modules=[m._to_serial() for m in self.modules],
+            extensions=[e._to_serial() for e in self.extensions],
+        )
+
+    def to_json(self) -> str:
+        return self._to_serial().model_dump_json()
+
+
+@dataclass
+class PackagePointer:
+    """Classes that point to packages and their inner contents."""
+
+    package: Package
+
+    def get_package(self) -> Package:
+        """Get the package pointed to."""
+        return self.package
```

Comment:
What is the purpose of this method?

Reply:
good point!

Reply:
(vestigial from original issue which proposed protocols)